be/src/connector/hive_connector.cpp

```diff
@@ -177,14 +177,37 @@ Status HiveDataSource::_init_partition_values() {
         }
 
         // eval conjuncts and skip if no rows.
-        RETURN_IF_ERROR(ExecNode::eval_conjuncts(_partition_conjunct_ctxs, partition_chunk.get()));
+        if (_has_scan_range_indicate_const_column) {
+            std::vector<ExprContext*> ctxs;
+            for (SlotId slotId : _scan_range.identity_partition_slot_ids) {
+                if (_conjunct_ctxs_by_slot.find(slotId) != _conjunct_ctxs_by_slot.end()) {
```

Comment:
why do we need to eval here, if it's identity partition column, iceberg sdk should be able to take care of it

Reply:
we only support some of the expr convert from starrock expr to iceberg expr, other is not supportted, eval here is for safety